File: Comunicación entre formularios Windows Forms/[C#]-Comunicación entre formularios Windows Forms/C#/DialogForm.cs
using System;
using System.Windows.Forms;

namespace DialogForm
{
    public partial class DialogForm : Form
    {
        public DialogForm()
        {
            InitializeComponent();
        }

        internal Person CurentPerson
        {
            get
            {
                return new Person()
                {
                    Name= txtName.Text,
                    Surname = txtSurname.Text,
                    Active = chkActive.Checked
                };
            }
            set
            {
                txtName.Text = value == null ? string.Empty : value.Name;
                txtSurname.Text = value == null ? string.Empty : value.Surname;
                chkActive.Checked = value != null && value.Active;
            }
        }

        private void btnOk_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.OK;
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
        }
    }
}

File: Comunicación entre formularios Windows Forms/[VB]-Comunicación entre formularios Windows Forms/VB/DialogForm.vb
Public Class DialogForm

    Friend Property CurrentPerson As Person
        Get
            Return New Person() With
            {
                .Name=txtName.Text,
                .Surname=txtSurname.Text,
                .Active=chkActive.Checked
            }
        End Get
        Set
            txtName.Text = If(Value Is Nothing, String.Empty, Value.Name)
            txtSurname.Text = If(Value Is Nothing, String.Empty, Value.Surname)
            chkActive.Checked = Value IsNot Nothing AndAlso value.Active
        End Set
    End Property

    Private Sub btnOk_Click(sender As Object, e As EventArgs) Handles btnOk.Click
        DialogResult = DialogResult.OK
    End Sub

    Private Sub btnCancel_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
        DialogResult = DialogResult.Cancel
    End Sub

End Class